### ISSUE TITLE
Mulitple DBContext / DbContextAttribute for migrations

### ISSUE THREAD

**Issue by generik0** [dotnet/efcore#24276]
Hi

We are creating a V2 application. Our V2 will be using EF Core 3+, our V1 uses EF Core 2.1.*
We cannot upgrade V2 due to breaking changes in EF 3+.

We need to share the Database between our V1 and V2 app. For Integration testing and also application migration.
We need to have 2 context configurations, but also need to run the same migrations with either context..  (Sharing DBSets via an abstract class.)

Currently the migrations are locked to a Context Type:

![image](https://user-images.githubusercontent.com/2411563/109193566-ca6ff000-7798-11eb-8e0d-9dadcc238e9a.png)

Is there any way to allow migrations to be unable for multiple contexts. E.g. in runtime we change what context can use the migrations?

EF Core 2->3 broke a lot due to the RelationalExtensions change. Hence, we have difficulty sharing context between versions :-/

Is there any way to achieve this? I.e. multiple contexts have access to a migration?

**Comment by bricelam** [comment]
@generik0 It's incredibly difficult to load both the 2.1 and 3.1 EF Core assemblies into the same process, and practically impossible to write source code against both versions in the same process. Could you provide more details on how you intend to use both versions in the same application?

Having said that, there may be certain advanced scenarios where adding two Migration attributes is valid, so I would certainly accept a PR that allowed multiple instances of it on a type.

**Comment by generik0** [comment]
Hi @bricelam 
Thanks for answering! :-). 
We don't intent to use two versions in same application.

- We have applicationV1 that
Is still on ef core V2.2
- We have applicationv2 that is on ef core 3.x

- the migrations are handles 1 place (currently v1) 

- We want to share the DB, and make sure all dbsets need to be common to both applications as they both need access to the same data.

- We do not want to have to copy paste db sets / mapping into the 2 difference DbContexts.

Basically we need to make sure that migrations and dbcontext is one place to ensure the integers of our very important data.

However, due to break ing changes, it is not possible.

- The configuration interface for modeling is different from v2 and v3
- The modeling extensions (like IsUnique()) is in an different extensions...


Any good ways to keep a common db + dbcontext + migration between a app running v2.c and another running v2.x

**Comment by ajcvickers** [comment]
@generik0 No, there is no good way to do that.
